client/config.c

```diff
@@ -26,6 +29,54 @@ TDNFConfGetRpmVerbosity(
     return nLogLevel;
 }
 
+static uint32_t TDNFParseOSInfo(PTDNF_CONF pConf)
+{
+    char buf[USERAGENT_HEADER_MAX_LENGTH];
+    char *name = NULL;
+    char *version = NULL;
+    uint32_t dwError = 0;
+    FILE *file = NULL;
+
+    file = fopen(OS_CONF_FILE, "r");
+
+    if (!file) {
+        pr_info("Warning: %s file is not present in the system\n", OS_CONF_FILE);
+        return 0;
+    }
+
+    while (fgets(buf, USERAGENT_HEADER_MAX_LENGTH, file))
+    {
+        if (strncmp("ID=", buf, sizeof("ID=")-1) == 0)
+        {
+            sscanf(buf, "ID=%s\n", buf);
```

Comment:
You need to take care of quotes, which may or may not be present. For example this is in Ubuntu:
```
$ cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
...
```
This is in Photon:
```
# cat /etc/os-release 
NAME="VMware Photon OS"
VERSION="5.0"
ID=photon
VERSION_ID=5.0
...
```
Note that Ubuntu has `VERSION_ID` in quotes, but Photon does not. You need to check for quotes and remove them if there.